src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Import/ImportOrchestratorJobTests.cs

```diff
@@ -68,16 +86,561 @@ public async Task GivenAnOrchestratorJobAndWrongEtag_WhenOrchestratorJobStart_Th
+            importOrchestratorInputData.InputFormat = "ndjson";
+            importOrchestratorInputData.InputSource = new Uri("http://dummy");
+            importOrchestratorInputData.RequestUri = new Uri("http://dummy");
+            JobInfo orchestratorJobInfo = (await testQueueClient.EnqueueAsync(0, new string[] { JsonConvert.SerializeObject(importOrchestratorInputData) }, 1, false, false, CancellationToken.None)).First();
+
+            integrationDataStoreClient.GetPropertiesAsync(Arg.Any<Uri>(), Arg.Any<CancellationToken>())
+                .Returns(callInfo =>
+                {
+                    Dictionary<string, object> properties = new Dictionary<string, object>();
+                    properties[IntegrationDataStoreClientConstants.BlobPropertyETag] = "test";
+                    properties[IntegrationDataStoreClientConstants.BlobPropertyLength] = 1000L;
+                    return properties;
+                });
+
+            sequenceIdGenerator.GetCurrentSequenceId().Returns(_ => 0L);
+
+            ImportOrchestratorJob orchestratorJob = new ImportOrchestratorJob(
+                mediator,
+                importOrchestratorInputData,
+                importOrchestratorJobResult,
+                contextAccessor,
+                fhirDataBulkImportOperation,
+                integrationDataStoreClient,
+                testQueueClient,
+                orchestratorJobInfo,
+                new Configs.ImportTaskConfiguration() { MaxRunningProcessingJobCount = 3 },
+                loggerFactory);
+            orchestratorJob.PollingFrequencyInSeconds = 0;
+            var jobExecutionException = await Assert.ThrowsAnyAsync<JobExecutionException>(() => orchestratorJob.ExecuteAsync(new Progress<string>(), CancellationToken.None));
+
+            ImportOrchestratorJobErrorResult resultDetails = (ImportOrchestratorJobErrorResult)jobExecutionException.Error;
+            Assert.Equal(HttpStatusCode.BadRequest, resultDetails.HttpStatusCode);
+            Assert.Equal(2, testQueueClient.JobInfos.Count(t => t.Status == JobStatus.Completed));
+            Assert.Equal(1, testQueueClient.JobInfos.Count(t => t.Status == JobStatus.Failed));
+            Assert.Equal(2, testQueueClient.JobInfos.Count(t => t.Status == JobStatus.Cancelled));
+
+            _ = mediator.Received().Publish(
+               Arg.Is<ImportJobMetricsNotification>(
+                   notification => notification.Id == orchestratorJobInfo.Id.ToString() &&
+                   notification.Status == JobStatus.Failed.ToString() &&
+                   notification.CreatedTime == importOrchestratorInputData.CreateTime &&
+                   notification.SucceedCount == 1 &&
+                   notification.FailedCount == 1),
+               Arg.Any<CancellationToken>());
+        }
+
+        [Fact]
+        public async Task GivenAnOrchestratorJob_WhenLastSubJobCancelld_ThenOperationCanceledExceptionShouldBeThrowAndWaitForOtherSubJobsCompleted()
```

Comment:
>WhenLastSubJobCancelld [](http://example.com/codeflow?start=49&length=22)

Since the cancel is group level, the orchestrator and process job would be cancelled at the same time.